src/Database/PostgreSQL/Transaction.hs

```diff
@@ -4,6 +4,7 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE ImplicitPrelude            #-}
 {-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE UndecidableInstances       #-}
```

Comment:
I would prefer to leave UndecidableInstances out of this, if at all possible. Down there lies madness.